File: python/2.py
#!/usr/bin/python

numbers = [1, 2]
max_value = 4000000
total = 2

while True:
  a = numbers[-1] + numbers[-2]
  if (a >= max_value):
    break

  numbers.append(a)
  if a % 2 == 0:
    total += a

print total

File: ruby/2.rb
#!/usr/bin/env ruby

$numbers = [1, 2]
$max_value = 4000000
$sum_value = 2

while ($numbers[-1] < $max_value) do
  $numbers.push($numbers[-1] + $numbers[-2])
  if $numbers[-1] %2 == 0
    $sum_value += $numbers[-1]
  end
end

puts $sum_value
